docs/AccessEnvironmentVariables.md

```diff
@@ -0,0 +1,174 @@
+
+Central configuration inversion name means that there is a single, centralized source of truth for configuration, and libraries use that source to accept configurations or not. This inverts the previous process, where configurations were defined in each library and we attempted to create a source of truth from the existing code, often leading to incomplete documentation.
+
+## Key Components
+
+### 1. Environment variables access
+
+Instead of accessing `ENV` directly, use `Datadog.get_environment_variable` to read environment variables:
+
+```ruby
+# ❌ Bad: Direct ENV access
+api_key = ENV['DD_API_KEY']
+timeout = ENV.fetch('DD_TIMEOUT', '30')
+
+# ✅ Good: Using Datadog.get_environment_variable
+api_key = Datadog.get_environment_variable('DD_API_KEY')
+timeout = Datadog.get_environment_variable('DD_TIMEOUT', '30')
+```
+
+This is enforced by the `CustomCops::EnvUsageCop` cop.
+
+### 2. Supported Configurations Registry
+
+All environment variables that start with `DD_` or `OTEL_` must be registered in the `supported-configurations.json` file to be accessible through `Datadog.get_environment_variable`.
+
+### 3. Automatic Code Enforcement
+
+Custom RuboCop cops automatically detect and prevent direct `ENV` usage:
+
+- `CustomCops::EnvUsageCop` - Prevents direct `ENV` access and auto-corrects to use `Datadog.get_environment_variable`
+- `CustomCops::EnvStringValidationCop` - Validates that environment variable strings are registered in supported configurations
+
+## How It Works
+
+The configuration inversion system works as follows:
+
+1. **Environment Variable Request**: Code calls `Datadog.get_environment_variable('DD_SOME_VAR')`
+2. **Validation**: The `ConfigHelper` checks if the variable is in the supported configurations
+3. **Access Control**:
+   - If the variable starts with `DD_` or `OTEL_` but is NOT in supported configurations → returns `nil` (or raises error in test environment)
+   - If the variable is supported or doesn't start with `DD_`/`OTEL_` and is not an alias → returns the value
+4. **Alias Resolution**: Checks for any configured aliases if the primary variable is not set
+5. **Deprecation Logging**: Logs deprecation warnings for deprecated variables
+
+## Adding New Environment Variables
+
+To add support for a new environment variable:
+
+### Step 1: Add to supported-configurations.json
+
+Edit the `supported-configurations.json` file and add your variable:
+
+```json
+{
+  "supportedConfigurations": {
+    "DD_YOUR_NEW_VARIABLE": {
+      "version": ["A"]
+    }
+  }
+}
+```
+
+#### Configuration Structure
+
+- **supportedConfigurations**: Maps variable names to configuration metadata
+  - `version`: (Currently always set to `["A"]`) Array indicating which tracer versions change the behavior of the configuration
```

Comment:
What does `A` mean? I understand it says "indicating which tracer versions change the behavior of the configuration", but what is `A`? An arbitrary anchor value? An alphabet sequence?
And should this be updated to `["A", "B"]` when the behaviour changes?

Without any docs, I'd imagine `version` could have the version of the tracer that started to support this env var (which maybe is just set to `2.19.0` today for all env vars in the JSON file).

Which makes me think: is this the tracer version, or env var semantic version? The way it is today, my best guess is that it's the semantic version of the env var itself (but it's just a guess).